app/controllers/criteria_controller.rb

```diff
@@ -28,11 +28,13 @@ def create
     end
     criterion_class = params[:criterion_type].constantize
     @criterion = criterion_class.new
-    @criterion.set_default_levels if params[:criterion_type] == 'RubricCriterion'
+
     if @criterion.update(name: params[:new_criterion_prompt],
                              assignment_id: @assignment.id,
                              max_mark: params[:max_mark_prompt],
                              position: @assignment.next_criterion_position)
+      @criterion.save!
```

Comment:
Calling `update` already saves the criterion, so there's no reason to save it again inside here (this is what I meant by it's redundant)